### PR TITLE
Feature/nric

### LINKED ISSUE #21: Implement NRIC functionality for patient details
Our management system needs to allow enter NRIC of each patient. Currently, NRIC serves as the identifier for the patients.

**Subtasks:**
- [x] Allow add NRIC commands to be parsed.
- [x] Implement basic methods for the `NRIC` class.
- [x] Aggregate `NRIC` attribute to `Person` class, add relevant methods.
- [x] Adjust relevant tests for `NRIC` component.

### PR DESCRIPTION
**Pull request by snajef**
resolves #21 

Ready for review.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -218,10 +236,8 @@ public boolean equals(Object other) {
             // state check
             EditPersonDescriptor e = (EditPersonDescriptor) other;
 
-            return getName().equals(e.getName())
-                    && getPhone().equals(e.getPhone())
-                    && getEmail().equals(e.getEmail())
-                    && getAddress().equals(e.getAddress())
+            return getNric().equals(e.getNric()) && getName().equals(e.getName()) && getPhone().equals(e.getPhone())
+                    && getEmail().equals(e.getEmail()) && getAddress().equals(e.getAddress())
```
Comment (yuntongzhang) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: /nit I feel that might me better to follow the original style here to make the code clearer.

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 2]
```diff
@@ -27,31 +29,35 @@
     /**
      * Parses the given {@code String} of arguments in the context of the AddCommand
      * and returns an AddCommand object for execution.
-     * @throws ParseException if the user input does not conform the expected format
+     *
+     * @throws ParseException
+     *             if the user input does not conform the expected format
```
Comment (yuntongzhang) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java]: /nit Similar to above. I think we should decide whether to keep @param on the same line or two different lines. Might need to check the coding style standard for cs2103.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -110,6 +138,7 @@ public static Tag parseTag(String tag) throws ParseException {
         return new Tag(trimmedTag);
     }
 
+    //@@author
```
Comment (yuntongzhang) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: Is this line intended?

File: src/main/java/seedu/address/model/person/Person.java [hunk 4]
```diff
@@ -53,30 +59,30 @@ public Address getAddress() {
...
      */
     public boolean isSamePerson(Person otherPerson) {
         if (otherPerson == this) {
             return true;
         }
 
-        return otherPerson != null
-                && otherPerson.getName().equals(getName())
+        return otherPerson != null && otherPerson.getNric().equals(getNric()) && otherPerson.getName().equals(getName())
                 && (otherPerson.getPhone().equals(getPhone()) || otherPerson.getEmail().equals(getEmail()));
```
Comment (yuntongzhang) [inline src/main/java/seedu/address/model/person/Person.java]: /nit Shall we put each `&&` condition on a separate line for clarity?

File: src/main/java/seedu/address/model/person/Person.java [hunk 5]
```diff
@@ -89,30 +95,22 @@ public boolean equals(Object other) {
...
 
         Person otherPerson = (Person) other;
-        return otherPerson.getName().equals(getName())
-                && otherPerson.getPhone().equals(getPhone())
-                && otherPerson.getEmail().equals(getEmail())
-                && otherPerson.getAddress().equals(getAddress())
-                && otherPerson.getTags().equals(getTags());
+        return otherPerson.getNric().equals(getNric()) && otherPerson.getName().equals(getName())
+                && otherPerson.getPhone().equals(getPhone()) && otherPerson.getEmail().equals(getEmail())
+                && otherPerson.getAddress().equals(getAddress()) && otherPerson.getTags().equals(getTags());
```
Comment (yuntongzhang) [inline src/main/java/seedu/address/model/person/Person.java]: /nit Same as above.

File: src/main/java/seedu/address/model/person/Person.java [hunk 6]
```diff
@@ -89,30 +95,22 @@ public boolean equals(Object other) {
...
-        builder.append(getName())
-                .append(" Phone: ")
-                .append(getPhone())
-                .append(" Email: ")
-                .append(getEmail())
-                .append(" Address: ")
-                .append(getAddress())
-                .append(" Tags: ");
+        builder.append(getNric()).append(" Name: ").append(getName()).append(" Phone: ").append(getPhone())
+        .append(" Email: ").append(getEmail()).append(" Address: ").append(getAddress()).append(" Tags: ");
```
Comment (yuntongzhang) [inline src/main/java/seedu/address/model/person/Person.java]: /nit Personally I feel the old style is clearer.

File: src/test/data/XmlAddressBookStorageTest/invalidAndValidPersonAddressBook.xml [hunk 7]
```diff
@@ -2,13 +2,15 @@
 <addressbook>
     <!-- Valid Person -->
     <persons>
+    	<nric>S1234567A</nric>
         <name>Hans Muster</name>
         <phone isPrivate="false">9482424</phone>
         <email isPrivate="false">hans@example.com</email>
         <address isPrivate="false">4th street</address>
```
Comment (yuntongzhang) [inline src/test/data/XmlAddressBookStorageTest/invalidAndValidPersonAddressBook.xml]: The person's attributes are not aligned here.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 8]
```diff
@@ -55,8 +58,10 @@
     private final EditPersonDescriptor editPersonDescriptor;
 
     /**
-     * @param index of the person in the filtered person list to edit
-     * @param editPersonDescriptor details to edit the person with
+     * @param index
+     *            of the person in the filtered person list to edit
+     * @param editPersonDescriptor
+     *            details to edit the person with
```
[the remark this answers is not in the document]
Comment (snajef) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Nope, it's just the default Eclipse formatter here. 
Do you want me to change it?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 3 above)
Comment (snajef) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Yeah, an empty //@@author tag is needed to indicate that the following methods aren't authored by me.

File: src/main/java/seedu/address/model/person/Person.java (hunk 4 above)
Comment (snajef) [reply src/main/java/seedu/address/model/person/Person.java]: Personally I don't think it's an issue, but I can see why you'd want it. Team decision?

File: src/main/java/seedu/address/model/person/Person.java (hunk 6 above)
Comment (snajef) [reply src/main/java/seedu/address/model/person/Person.java]: Oh it's just the default Eclipse formatter.

Do you want me to revert this?

File: src/test/data/XmlAddressBookStorageTest/invalidAndValidPersonAddressBook.xml (hunk 7 above)
Comment (snajef) [reply src/test/data/XmlAddressBookStorageTest/invalidAndValidPersonAddressBook.xml]: My bad, I'll fix this in a future commit.

File: src/main/java/seedu/address/model/person/Person.java (hunk 6 above)
Comment (yuntongzhang) [reply src/main/java/seedu/address/model/person/Person.java]: Yep I think put a chaining method to a different line looks cleaner.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 3 above)
Comment (yuntongzhang) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Ic, thanks!

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (yuntongzhang) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Yep, I think we should follow the original style in the codebase.

[the remark this answers is not in the document]
Comment (yuntongzhang) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Final decision: @param description should be on the same line as the @param itself.

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 1 above)
Comment (yuntongzhang) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Final decision: split conditions (one condition one line).

**Pull request merged**